infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/endpoints/JavalinRestApiRequest.java

```diff
@@ -244,6 +246,37 @@ public <T> List<T> getQueryParameterList(final ParameterMetadata<T> parameterMet
     }
   }
 
+  @Override
+  public <T> Optional<T> getOptionalHeader(final ParameterMetadata<T> parameterMetadata) {
+    if (!headerMap.containsKey(parameterMetadata.getName())) {
+      return Optional.empty();
+    }
+    try {
+      return Optional.of(
+          parameterMetadata
+              .getType()
+              .deserializeFromString(headerMap.get(parameterMetadata.getName())));
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Could not read optional header %s: %s",
+              parameterMetadata.getName(), e.getMessage()));
+    }
+  }
+
+  @Override
+  public <T> T getHeader(final ParameterMetadata<T> parameterMetadata) {
+    try {
+      return parameterMetadata
+          .getType()
+          .deserializeFromString(headerMap.get(parameterMetadata.getName()));
```

Comment:
If the headerMap does not contain the key that we are looking for it will throw a NullPointerException. This is an example of a bug that would be caught with unit tests.

Reply:
Methods like `getPathParameter` would have a similar issue. Do I need to update those to check for this too?